### ISSUE TITLE
Patched Promise swallows exceptions.

### ISSUE THREAD

**Issue by NikGovorov** [angular/zone.js#239]
Please find an example here: http://plnkr.co/edit/dEmR53zIRBx4e3BINi9P?p=preview, removing zone.js reference from index.html fixes the case.

Found in a typical Angular 2 app use case:

```typescript
async ngOnInit() {
    let data = await this._service.load();
    throw new Error('Error in ngOnInit'); // will be never reported, even in console
}
```

Thoughts?

**Comment by robwormald** [comment]
i'm reasonably sure this is the expected behavior with async/await - see https://jsfiddle.net/robwormald/k1oLf6qu/1/

its effectively throwing at your first line - the async function returns a promise, which is where the error is caught (which of course you have no access to, as its a lifecycle method) - you should generally use try/catch with async/await.

**Comment by NikGovorov** [comment]
http://plnkr.co/edit/dEmR53zIRBx4e3BINi9P?p=preview - demonstrates the issue without async/await(unfortunately plunker is currently down).

In your case you're right, but I didn't say  that ```let data = await this._service.load();```  fails/rejects in my sample. I forked your example to demonstrate the issue: https://jsfiddle.net/v9vyj5n0/2/, removing ```angular2-polyfills.min.js``` which includes ```zone.js``` fixes the issue.

**Comment by NikGovorov** [comment]
It seems to be fixed in 0.6.4
